src/lib/Persistence/Legacy/Content/URLWildcard/Query/CriterionHandler/SourceUrl.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace Ibexa\Core\Persistence\Legacy\Content\URLWildcard\Query\CriterionHandler;
+
+use Doctrine\DBAL\ParameterType;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Ibexa\Contracts\Core\Repository\Values\Content\URLWildcard\Query\Criterion;
+use Ibexa\Core\Persistence\Legacy\Content\URLWildcard\Query\CriteriaConverter;
+use Ibexa\Core\Persistence\Legacy\Content\URLWildcard\Query\CriterionHandler;
+
+final class SourceUrl implements CriterionHandler
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function accept(Criterion $criterion): bool
+    {
+        return $criterion instanceof Criterion\SourceUrl;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(
+        CriteriaConverter $converter,
+        QueryBuilder $queryBuilder,
+        Criterion $criterion
+    ) {
+        /** @var \Ibexa\Contracts\Core\Repository\Values\Content\URLWildcard\Query\Criterion\SourceUrl $criterion */
+        return $queryBuilder->expr()->like(
+            'source_url',
+            $queryBuilder->createNamedParameter(
+                '%' . $criterion->sourceUrl . '%',
```

Comment:
Have you tested this for URLs containing URI escape chars, like `%20`?

Reply:
Added test for this, and working correctly, I also tested manually and working correctly